src/TLuaInterpreter.cpp

```diff
@@ -18171,6 +18174,52 @@ int TLuaInterpreter::getDictionaryWordList(lua_State* L)
     return 1;
 }
 
+int TLuaInterpreter::getPlayerName(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+
+    lua_pushstring(L, host.getLogin().toUtf8().constData());
+    return 1;
+}
+
+int TLuaInterpreter::sendPlayerName(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+
+    if (host.getLogin().isEmpty()) {
+        lua_pushnil(L);
+        lua_pushstring(L, "no character name set so nothing to send");
+        return 2;
+    } else {
+        host.mTelnet.slot_send_login();
+        lua_pushboolean(L, true);
+        return 1;
+    }
+}
+
+int TLuaInterpreter::sendPlayerPassword(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+
+    // This is a tiny bit of information leak but give the overall security of
```

Comment:
The information leak is that the player has not set a password - at least not stored in the in-application system. And the only people who are going to see that comment are going to looking at the implementation details anyhow.

OTOH I suppose it would not be unreasonable to put the time out test first so that this message is only going to be seen by people trying to debug things (why can't I log in - oh, I didn't enter a password/oh, perhaps I shouldn't have deleted that file in the profile directory)...

A shorter error message for the other (timeout) check I suppose is not unreasonable.
So from:
`password sending disabled, outside of period that follows making a successful connection to game-server`
to:
`sending the password is only allowed for a short time after connecting to the game`

Reply:
In next commit.